### PR TITLE
Use terraform to manage preview environment certificates 

### PR DESCRIPTION
**Pull request by mads-hartmann**
## Description

Use Terraform to request certificates for preview environment instead of CertManager. It uses the ACME Provider to integrate with Letsencrypt and ZeroSSL.

**Thought**: Currently I create an acme_registration with a unique private key for each preview environment. This might be a bit overkill, alternatively we could have the acme_registration resources in gitpod-io/ops instead - the down-side of this is that we'd then have to store the pem somewhere so we can consume it in this repository when we create the certificate resource. From my experience it doesn't take a long time (less then 10 seconds) to create these resources, so right now I'm leaning towards keeping them here, unless there are any other technical reasons we wouldn't want to create an acme_registration for each preview environment.

**Follow up after this PR** 
- Delete the cluster issuers from core-dev
- Delete cert manager in core-dev
- Update [Debugging certificate issues](https://www.notion.so/gitpod/Debugging-certificate-issues-9453d1c8ac914ce7962557b67f7b49b3) as the existing instructions are for CertManager

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/15146

## How to test

I verified that new preview environments created with these changes got valid certificates:

```
# I used unique branches for each of these to get a new preview environment name.

# Letsencrypt
leeway run dev:preview
leeway run dev/preview:delete-preview

# ZeroSSL
TF_VAR_cert_issuer=zerossl-issuer-gitpod-core-dev leeway run dev:preview
TF_VAR_cert_issuer=zerossl-issuer-gitpod-core-dev leeway run dev/preview:delete-preview
```

## Release Notes

```release-note
NONE
```

## Documentation

N/A

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by mads-hartmann** [comment]
I successfully tested using ZeroSSL now. That means both Letsencrypt and ZeroSSL has been tested ☺️ 

Screenshot 2023-01-03 at 15 54 27

**Review by meysholdt — APPROVED** [review]
Code change LGTM! 
But I think @vulkoingim should review this, too.
And we should merge it at the beginning of the day so that we're available to support colleagues in case anything is bumpy.

**Review by vulkoingim — APPROVED** [review]

File: dev/preview/infrastructure/harvester/certificate-letsencrypt.tf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  count           = local.letsencrypt_enabled ? 1 : 0
+  provider        = acme.letsencrypt
+  account_key_pem = tls_private_key.letsencrypt[0].private_key_pem
+  email_address   = "preview-environment-certificate-throwaway@gitpod.io"
+}
+
+resource "acme_certificate" "letsencrypt" {
+  count           = local.letsencrypt_enabled ? 1 : 0
+  provider        = acme.letsencrypt
+  account_key_pem = acme_registration.letsencrypt[0].account_key_pem
```
[the remark this answers is not in the document]
Comment (mads-hartmann) [reply dev/preview/infrastructure/harvester/certificate-letsencrypt.tf:20]: @vulkoingim If I do that then I re-use the accounts the cert-manager uses - for now I'd like to keep them separate ☺️ If we run into rate-limits for creating accounts then we can do it like this ☺️

[the remark this answers is not in the document]
Comment (vulkoingim) [reply dev/preview/infrastructure/harvester/certificate-letsencrypt.tf:20]: If I remember correctly, the rate limit is per domain. But doesn't hurt to leave it as is. Feel free to get it merged

**Pull request merged**